render/egl.c

```diff
@@ -359,13 +364,93 @@ struct wlr_egl *wlr_egl_create_with_drm_fd(int drm_fd) {
 		}
 	}
 
-	return egl;
+	return true;
+}
 
+static bool device_has_name(const drmDevice *device, const char *name);
+
+struct wlr_egl *wlr_egl_create_with_drm_fd(int drm_fd) {
+	struct wlr_egl *egl = egl_create();
+	if (egl == NULL) {
+		wlr_log(WLR_ERROR, "Failed to create EGL context");
+		return NULL;
+	}
+
+	if (egl->procs.eglQueryDevicesEXT == NULL) {
+		wlr_log(WLR_ERROR, "EGL_EXT_device_enumeration not supported");
+		goto error;
+	}
+
+	EGLint nb_devices = 0;
```

Comment:
Maybe we can move all of this `EGLDevice` logic into a function among these lines?

```c
static EGLDevice get_egl_device_from_drm_fd(struct wlr_egl *egl, int drm_fd)
```